ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+      response: (data, _matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          towerOnYou: {
+            en: '${color} Tower on You (w/ ${partner})',
+          },
+          towerLater: {
+            en: '${color} Towers later',
+          },
+          light: {
+            en: 'Light',
+          },
+          dark: {
+            en: 'Dark',
+          },
+          unknown: Outputs.unknown,
+        };
+        const towerOnYou = data.engravement3TowerPlayers.indexOf(data.me) !== -1;
+        let towerColor = output.unknown!();
+        if (data.engravement3TowerType !== undefined)
+          towerColor = data.engravement3TowerType === 'lightTower'
+            ? output.light!()
+            : output.dark!();
+        let partner;
+        if (towerOnYou) {
+          partner = data.ShortName(data.engravement3TowerPlayers.find((name) =>
+            name !== data.me
+          )) ?? output.unknown!();
+          return { alertText: output.towerOnYou!({ color: towerColor, partner: partner }) };
+        } else if (data.party.isDPS(data.me)) {
+          // DPS want to know tower color to know if they are soaking or baiting add cleaves later
+          return { infoText: output.towerLater!({ color: towerColor }) };
+        }
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Towers Drop',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTowerIds },
+      condition: (data, matches) => data.engravementCounter === 3 && data.me === matches.target,
+      delaySeconds: (_data, matches) => parseFloat(matches.duration) - 5,
+      alertText: (data, _matches, output) => {
+        let towerColor = output.unknown!();
+        if (data.engravement3TowerType !== undefined)
+          towerColor = data.engravement3TowerType === 'lightTower'
+            ? output.light!()
+            : output.dark!();
+        return output.dropTower!({ color: towerColor });
+      },
+      outputStrings: {
+        dropTower: {
+          en: 'Drop ${color} Tower',
+        },
+        light: {
+          en: 'Light',
+        },
+        dark: {
+          en: 'Dark',
+        },
+        unknown: Outputs.unknown,
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Soak Tower/Bait Adds',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTiltIds },
+      condition: (data, matches) => data.engravementCounter === 3 && data.me === matches.target,
+      suppressSeconds: 15, // avoid second (incorrect) alert when debuff switches from soaking tower
+      alertText: (data, matches, output) => {
+        // lightTower/darkTower support players receive lightTilt/darkTilt once dropping their tower
+        // so exclude them from receiving this alert
+        if (data.engravement3TowerPlayers.indexOf(data.me) !== -1)
+          return;
+
+        const soakMap: { [key: string]: string } = {
+          lightTower: 'darkTilt',
+          darkTower: 'lightTilt',
+        };
```

Comment:
```suggestion
        const soakMap = {
          lightTower: 'darkTilt',
          darkTower: 'lightTilt',
        } as const;
```

If `engravement3TowerType` is more strongly typed, then this can just be `as const`.